### ISSUE TITLE
crash in esp_yield

### ISSUE THREAD

**Issue by marcmerlin** [esp8266/Arduino#4675]
### Basic Infos

- [ x] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [ x] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [ ] I have tested that the issue is present in current master branch (aka latest git).
- [ x] I have searched the issue tracker for a similar issue.
- [ x] If there is a stack dump, I have decoded it.
- [ x] I have filled out all fields below.

#### Platform

- Hardware:				ESP-12 (wemos D1 mini)
- Core Version:			2.3.0-rc2
- Development Env:		Arduino IDE
- Operating System:		linux/debian

### Settings in IDE

- Module:  			Wemos D1 mini r2
- Flash Mode:			?
- Flash Size: 		4MB
- lwip Variant:   ?
- Reset Method:		?
- Flash Frequency:	?
- CPU Frequency:		80Mhz
- Upload Using:	SERIAL
- Upload Speed:   115200

### Problem Description

I have other FastLED code working on that board, but this new sketch (which I never used to ESP8266 because it would not work until I converted it to PROGMEM), is causing a weird traceback.

### [MCVE]
https://github.com/marcmerlin/NeoMatrix-FastLED-IR/blob/master/.test8flower.ino

### Debug Messages

0x402043fc: esp_yield() at /home/merlin/.arduino15/packages/esp8266/hardware/esp8266/2.3.0-rc2/cores/esp8266/core_esp8266_main.cpp line 86
0x40202130: TimerEvent() at /home/merlin/prg/arduino/Neopixel/NeoMatrix-Anim/test8flower/test8flower.ino line 3119
0x402010ae: delay at /home/merlin/.arduino15/packages/esp8266/hardware/esp8266/2.3.0-rc2/cores/esp8266/core_esp8266_wiring.c line 53
0x4020332a: loop() at /home/merlin/prg/arduino/Neopixel/NeoMatrix-Anim/test8flower/test8flower.ino line 3113
0x40204448: loop_wrapper() at /home/merlin/.arduino15/packages/esp8266/hardware/esp8266/2.3.0-rc2/cores/esp8266/core_esp8266_main.cpp line 123

Serial debug output:
Num Pixels: 768

Exception (3):
epc1=0x40202182 epc2=0x00000000 epc3=0x00000000 excvaddr=0x40247628 depc=0x00000000

ctx: cont 
sp: 3fff0030 end: 3fff0220 offset: 01a0

>>>stack>>>
3fff01d0:  3fffdad0 3ffef1f0 40204610 40202164  
3fff01e0:  402010ae 00000064 3ffee6f4 3ffef1f0  
3fff01f0:  3fffdad0 00000000 3ffef1e8 40203362  
3fff0200:  feefeffe feefeffe 3ffef1e8 4020465c  
3fff0210:  feefeffe feefeffe 3ffef200 40100718  
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(1,6)

 ets Jan  8 2013,rst cause:4, boot mode:(1,6)

wdt reset

**Comment by devyte** [comment]
Oh boy
1. Why are you using core 2.3.0-rc2? It is beyond ancient, older than even release 2.3.0, and it has so many issues, it's beyond reason.
2. The linked sketch, while not too large code-wise, is in fact huge due to the embedded data. How is it a _MINIMAL_ sketch?
3. The sketch is a thin app over a 3rd party lib. The issue template early states that support for use of 3rd party libs should be sought elsewhere, like a  community forum. This is an issue tracker for the core code.
4. The sketch has all sorts of wrong in it. For example,  in one loop pass it attempts to process all frames at once, and there are no yields in between. I would not be surprised at all to see the wdt triggering all over the place.

I don't see any indication of a core issue. Closing.

**Comment by marcmerlin** [comment]
Thanks for your answers and sorry I didn't get some of the points right, as you said.
First, I can report that the problem is fixed now by upgrading to 2.4.1.

Given that the crash seemed to be inside the esp8266 core library, I thought that was a problem  (as opposed to a watchdog reset which is my fault)
I'm not asking for help with libraries or code, which otherwise work well on other arch like teensy 3.1.
Yes, the code is crap, I didn't write it, it came from a code generator. Your point about yield not being there often enough is fair, thank you, although it was not a yield issue in the end.

As for the very old 2.3.0-rc2, this may be a problem that could impact others. I'll give details in case it helps anyone who stumbles on this.
1) 2.3.0-rc2 was sadly the latest available in the latest arduino IDE
Mmmh, I see my arduino had 
http://arduino.esp8266.com/staging/package_esp8266com_index.json
from a long time ago, instead of
http://arduino.esp8266.com/stable/package_esp8266com_index.json
So now, I have 2.4.1 and it doesn't compile the code anymore because the progmem interface changed, but once I adjusted to it, things work now, thank you.

**Comment by marcmerlin** [comment]
@devyte 
Sorry again for my issues due to having an old staging URL in my arduino config. Maybe it would be good to remove/invalidate it if it's not going to track up to date releases anymore?

And yes, thanks for the warning, and I am indeed quite aware of timing issues on ESP8266 and that I can't expect that code will "just work". I've ported some code to ESP8266, and more code (and libraries) to ESP32 http://marc.merlins.org/perso/arduino/post_2017-01-16_IoTuz-Driver-for-our-ESP32-board-built-at-Open-Hardware-Miniconf-at-Linux_Conf_au-2017.html
Doing 4 parallel neopixel output on ESP8266 with FastLED is challenging, but possible, I got it working :)
https://plus.google.com/u/0/communities/109127054924227823508
